drivers/sensor/stm32_temp/stm32_temp.c

```diff
@@ -15,9 +15,8 @@ LOG_MODULE_REGISTER(stm32_temp, CONFIG_SENSOR_LOG_LEVEL);
 #define CAL_RES 12
 #if DT_HAS_COMPAT_STATUS_OKAY(st_stm32_temp)
 #define DT_DRV_COMPAT st_stm32_temp
-#define HAS_CALIBRATION 0
-#elif DT_HAS_COMPAT_STATUS_OKAY(st_stm32_temp_cal)
-#define DT_DRV_COMPAT st_stm32_temp_cal
+#elif DT_HAS_COMPAT_STATUS_OKAY(st_stm32_temp_cal_v2)
+#define DT_DRV_COMPAT st_stm32_temp_cal_v2
 #define HAS_CALIBRATION 1
```

Comment:
As you're onto, I did look at what scheme had been used for other drivers and saw the v1, v2 being used for I2C.

I totally agree with you, naming the new version "st,stm32c0-temp-cal" makes much more sense, thanks for the input. 👍

Reply:
Interestingly, the C0 series does not seem to be the first one to have only one temperature calibration register.
The datasheets for the STM32F0x0 indicates one also (but that does not transpire in the reference manual).

I've tried to go through all other series and this is the only one I found. I remember having seen this peculiarity when reworking the dtsi include tree for the F0 family.

Reply:
Interesting, does it fit with the changes made here? I don't have a stm32f0x0 board to test with.

Reply:
To have it work on F0x0, we would need to add another property in st,stm32-temp-cal-c0.yaml: `ntc`, just as in st,stm32-temp.yaml.
For C0 we subtract the calibrated value to the measured one, for F0x0, it is the other way around.
For C0:
![image](https://user-images.githubusercontent.com/103205175/231106343-7ca8a688-f7d0-4329-a6e8-3ca7a5e8092b.png)
For F0x0:
![image](https://user-images.githubusercontent.com/103205175/231106596-dadf6c5d-3432-4692-84d2-6c9c6e20237b.png)
Otherwise, the calculations are identical.